themes/src/main/resources/theme/base/admin/resources/js/controllers/realm.js

```diff
@@ -556,10 +556,14 @@ module.controller('RealmLocalizationCtrl', function($scope, Current, $location,
 
     $scope.updateRealmSpecificLocalizationTexts = function() {
         RealmSpecificLocalizationTexts.get({id: realm.realm, locale: $scope.selectedRealmSpecificLocales }, function (updated) {
-            $scope.localizationTexts = updated;
+            $scope.localizationTexts = sortObjectByKeys(updated);
         })
     }
 
+    function sortObjectByKeys(o) {
+        return Object.keys(o).sort().reduce((r, k) => (r[k] = o[k], r), {});
```

Comment:
Arrow functions are ES2015 syntax, and to my knowledge we're not transpiling this code to older versions of JavaScript. Since we still have to support older browsers for the Admin UI I believe this could break the code there.

There is also no guarantee that the properties of an object will remain in insertion order when iterating on them. See the definition of an [Object from ECMAScript Third Edition (pdf)](http://www.ecma-international.org/publications/files/ECMA-ST-ARCH/ECMA-262,%203rd%20edition,%20December%201999.pdf):

> ## 4.3.3 Object
> 
> An object is a member of the type Object. **It is an unordered collection of properties** each of which contains a primitive value, object, or function. A function stored in a property of an object is called a method.

This means that we would have to use some kind of type here like Array or Set to ensure that this remains in the correct order.

Reply:
Hi @jonkoops - thanks for looking at this and finding this I didn't. "+1" for the arrow functions. 

As I was wondering why my manual test succeeded while the spec says this is no guaranteed behavior, I found this discussion on SO indicating that the specified behavior changed in ES6 (see answer by Mathias Bynens): https://stackoverflow.com/questions/5467129/sort-javascript-object-by-key

As you point out that this should run on older browsers as well, this must not be relied upon. A quick look at the docs didn't reveal a list of supported browsers to me. Could you please point me at more information about this topic? Thanks!

Reply:
I do not believe we currently have the browser support publicly documented, but the worst case browser we have to support at the moment is Internet Explorer 11. For the rest of the browsers the target is pretty evergreen.

In the new Admin UI we're targeting only evergreen browsers.

Reply:
Thank you so much for your feedback.
I was not aware of these constraints and updated the PR to not use an Object anymore.
Since ng-repeat does not support Map iteration, I used an Array instead.
Further feedback is very welcome.

Reply:
Great! I think using an array here is an excellent option considering the browser constraints.